gensim/models/fasttext.py

```diff
@@ -12,74 +34,189 @@
 
 logger = logging.getLogger(__name__)
 
-MAX_WORDS_IN_BATCH = 10000
+try:
+    from gensim.models.fasttext_inner import train_batch_sg, train_batch_cbow
+    from gensim.models.fasttext_inner import FAST_VERSION, MAX_WORDS_IN_BATCH
+    logger.debug('Fast version of Fasttext is being used')
+
+except ImportError:
+    # failed... fall back to plain numpy (20-80x slower training than the above)
+    logger.warning('Slow version of Fasttext is being used')
+    FAST_VERSION = -1
+    MAX_WORDS_IN_BATCH = 10000
+
+    def train_batch_cbow(model, sentences, alpha, work=None, neu1=None):
+        """
+        Update CBOW model by training on a sequence of sentences.
```

Comment:
Please use numpy style docstrings (here and everywhere), useful links: http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html, https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt

Reply:
Sure, will make the changes.